src/cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -139,16 +82,41 @@ extern "C" COCOTB_EXPORT void _embed_init_python(void) {
         }
     }
 
-    to_python();
-    // must set program name to Python executable before initialization, so
-    // initialization can determine path from executable
-    set_program_name_in_venv();
-    Py_Initialize(); /* Initialize the interpreter */
-    PySys_SetArgvEx(1, argv, 0);
```

Comment:
AFAICT setting the argv to `["cocotb"]` does nothing. Ostensibly Python sees that it's a "script" and adds the parent path to `sys.path`, but the cocotb module's parent is already on the path since it is an installed module since cocotb 1.2 or something.

Reply:
I think the main benefit here is that user code can look at `sys.argv` and determine (in a suboptimal way) that they are running inside cocotb; I think what you describe doesn't happen because `updatepath` (the last argument) is `0`.